src/app/components1/edit-product/edit-product.component.ts

```diff
@@ -0,0 +1,82 @@
+import { Component, OnInit } from '@angular/core';
+import {
+  FormBuilder,
+  FormGroup,
+  FormControl,
+  Validators,
+} from '@angular/forms';
+import { ActivatedRoute, Router } from '@angular/router';
+import { ProductService } from '../product.service';
+import { HttpClient } from '@angular/common/http';
+import Product from '../Types/Product';
+import { Location } from '@angular/common';
+@Component({
+  selector: 'app-edit-product',
+  templateUrl: './edit-product.component.html',
+  styleUrls: ['./edit-product.component.scss'],
+})
+export class EditProductComponent implements OnInit {
+  product: Product = {
+    id: 0,
+    description: '',
+    name: '',
+    category: '',
+    price: 0,
+    image: '',
+  };
+
+  id = Number(this.route.snapshot.paramMap.get('id'));
+  profileForm = this.fb.group({
+    name: ['', [Validators.required,Validators.nullValidator]],
+    category: ['', [Validators.required,Validators.nullValidator]],
+    price: [0, [Validators.required, Validators.min(1)]],
+    //image: ['', [Validators.required,Validators.nullValidator]],
+    description: ['', [Validators.required,Validators.nullValidator]],
+  });
+
+  constructor(
+    private fb: FormBuilder,
+    private route: ActivatedRoute,
+    private location: Location,
+    private productService: ProductService,
+    private http: HttpClient,
+    private router:Router
+  ) {}
+
+  ngOnInit(): void {
+    this.http.get<Product>(`${'http://localhost:3000'}/products/${this.id}`).subscribe((data)=>{
```

Comment:
**suggestion:** This also should be in a products service. Moreover, you cand keep this url inside the `environment.ts` file or you could use a [proxy](https://angular.io/guide/build#proxying-to-a-backend-server).